torchvision/utils.py

```diff
@@ -315,7 +318,10 @@ def draw_segmentation_masks(
         img_to_draw[:, mask] = color[:, None]
 
     out = image * (1 - alpha) + img_to_draw * alpha
-    return out.to(out_dtype)
+    if original_dtype in {torch.float16, torch.float32, torch.float64}:
```

Comment:
Check for `is_floating_point()` instead.

Reply:
Hey there! @NicolasHug 

We can use to_dtypes as mentioned above but there might be two issues with it:
1) `from torchvision.transforms.v2.functional import to_dtype` when imported normally (globally) gives us circular import due to the `_log_api_usage_once(to_dtype)` which is being imported in the to_dtype function, to overcome this we can either import in inside `draw_segmentation_masks()` method or we manually convert it ourselves.

2) When float dtype image is passed converting it back to float like so 
```
out = image * (1 - alpha) + img_to_draw * alpha
if original_image.is_floating_point():
    out = to_dtype(out, torch.float, scale=True)
```

is resulting in mismatched elements for the unit test, and is only working when we divide it with 255.0
```
out = image * (1 - alpha) + img_to_draw * alpha
if original_image.is_floating_point():
    out = to_dtype(out, torch.float) / 255.0
```

Thoughts on this?
Appreciate your time, thank you!

Reply:
Hello @NicolasHug,

I hope this message finds you well. I wanted to seek your advice regarding the usage of the is_floating_point() method. In order to utilize it, one option is to pass in a tensor as follows:

`torch.is_floating_point(torch.as_tensor(0, dtype=original_dtype))`

Alternatively, we could clone the image `image.clone()` and check the cloned image as `clone_image.is_floating_point()`

I would appreciate your suggestion on the most appropriate approach or if there is any other recommended way to proceed in this context.